src/Microsoft.Azure.WebJobs/ExtensionAttribute.cs

```diff
@@ -6,19 +6,25 @@
 namespace Microsoft.Azure.WebJobs.Description
 {
     /// <summary>
-    /// Attribute applied to the <see cref="IExtensionConfigProvider"/> implementation for an extension.
+    /// Attribute used to mark types belonging to a single logical extension.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class, Inherited = false)]
     public sealed class ExtensionAttribute : Attribute
     {
-        public ExtensionAttribute(string name)
+        public ExtensionAttribute(string name, string configName = null)
         {
             Name = name;
+            ConfigName = configName ?? name;
         }
 
         /// <summary>
         /// Gets the friendly name of the extension.
         /// </summary>
         public string Name { get;  }
+
+        /// <summary>
+        /// Gets the configuration section name for this extension.
+        /// </summary>
+        public string ConfigName { get; }
```

Comment:
Should this be more specific? `ConfigurationSection`?

Reply:
Fixed